src/search/landmarks/landmark_count_heuristic.cc

```diff
@@ -199,30 +201,19 @@ int LandmarkCountHeuristic::compute_heuristic(const State &ancestor_state) {
     int h = get_heuristic_value(ancestor_state);
 
     if (use_preferred_operators) {
-        BitsetView landmark_info = lm_status_manager->get_reached_landmarks(ancestor_state);
-        LandmarkNodeSet reached_lms = convert_to_landmark_set(landmark_info);
-        generate_helpful_actions(state, reached_lms);
+        generate_helpful_actions(state);
     }
 
     return h;
 }
 
-bool LandmarkCountHeuristic::check_node_orders_disobeyed(const LandmarkNode &node,
-                                                         const LandmarkNodeSet &reached) const {
-    for (const auto &parent : node.parents) {
-        if (reached.count(parent.first) == 0) {
-            return true;
-        }
-    }
-    return false;
-}
-
-bool LandmarkCountHeuristic::generate_helpful_actions(
-    const State &state, const LandmarkNodeSet &reached) {
-    /* Find actions that achieve new landmark leaves. If no such action exist,
-     return false. If a simple landmark can be achieved, return only operators
-     that achieve simple landmarks, else return operators that achieve
-     disjunctive landmarks */
+bool LandmarkCountHeuristic::generate_helpful_actions(const State &state) {
```

Comment:
We should clean up the signature: there is no longer a reason to have this be a non-void function. The general preference is for mutating functions like these ones to be void; the return value is not obvious from the name and is unused. Also, the function name should be generate_preferred_operators; "helpful actions" is an EHC/FF term. If we make this void, the following comment also should be adjusted.

Reply:
This also entails changing the variable names that use "ha".